content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,573 @@
+ 
+The next step is to load the data into the notebook:
+
+```py
+df = pd.read_csv("/content/drive/MyDrive/IRIS.csv")
+df.columns =['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'species']
+```
+
+```py
+df.head()
+```
+
+![dataset head](/engineering-education/statistical-hypothesis-testing-python-implementation/dataset_head.png)
+
+let's run df.sample() to visualize samples randomly.
+
+```python
+df.sample(10)
+```
+
+![sampleten](/engineering-education/statistical-hypothesis-testing-python-implementation/dataset_sample.png)
+ 
+Always check for missing values to avoid errors.
+
+```py
+df.isnull().sum()
+```
+
+```bash
+sepal_length    0
+sepal_width     0
+petal_length    0
+petal_width     0
+species         0
+dtype: int64
+```
+
+```py
+print(df["species"].unique())
+```
+
+```bash
+array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)
+```
+
+We can be sure now that there are 3 classes of species and there are no missing values.
+Let's proceed with the data manipulation.
+ 
+Let's use `petal_width` to compare species for our Chi-square test.
+ 
+The first step is to check the summary of `petal_width`. We then use this information to convert it to a categorical variable.
+
+```python
+print(df.petal_width.describe())
+```
+
+Output:
+
+```bash
+count    150.000000
+mean       1.198667
+std        0.763161
+min        0.100000
+25%        0.300000
+50%        1.300000
+75%        1.800000
+max        2.500000
+Name: petal_width, dtype: float64
+```
+
+The Chi-square test helps in determining if there is a notable difference between observed and normal frequencies in one or more categories. 
+
+The values of `petal_width` must be changed into categories of zeros and ones using the 50% percentile.
+
+There are two conditions that must be satisfied while calculating percentile:
+1. Firstly, data must be arranged in an ascending order.
+2. Secondly, the index must be rounded up if it is not an integer.
+
+```py
+print(np.percentile(df.petal_width, 50))
+```
+
+Output:
+
+```bash
+1.3
+```
+
+A quartile is another term relating to percentile. You can learn more about percentiles and quartiles from [here](https://datascienceupskill.com/statistics/percentiles-and-quartiles). 
+ 
+```py
+def petal_cat(df):
+  if df['petal_width'] <= 1.3:
+    return 0
+  elif df['petal_width'] > 1.3:
+    return 1
+  else:
+    return 'indifferent'
+ 
+df['petal_width_new'] = df.apply(petal_cat, axis=1) 
+```
+
+The  data format for the `species` class has to be transformed to 0, 1, and 2, which is a machine-readable format to facilitate testing. 
+
+Therefore, with [label-encoding](/engineering-education/introduction-to-scikit-learn-in-python/), a numeric value will be assigned to each of the categories in the class. 
+
+To learn more about label encoding, [click](/engineering-education/data-preprocessing-in-r/).
```

Comment:
Change the sentence to -
"To learn more about label encoding, the reader is encouraged to go through the following [resource](/engineering-education/data-preprocessing-in-r/)."